### PR TITLE
Creating podcasts by users

### LINKED ISSUE #3537: Podcast ownership
Part of the Podcast section enrichment #271 
>  Users should be able to act as admins for podcasts

- create a `podcast_admin` role
- add appropriate roles to existing users (how?)
- assign existing podcasts to users when possible

- creating podcasts by users
- editing podcasts by podcast admins
- notify a podcast admin if a podcast was not properly fetched

- podcasts in the user's profile
- podcasts on user's dashboard

I don't see the need to necessarily create an organization for each podcast right now.
If you have other thoughts, please share.

A couple of more concerns:
- I can imagine that users who don't own podcasts may want to add a podcast too. We may need to add such a possibility too.
- I'm not sure where the list of the user's podcasts (for managing) should be.

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
I've started implementing a feature for users to be able to suggest podcasts.

What else needs to be done before merging:

- [x] validate `feed_url`
- [x] validate `main_color_hex`
- [x] provide form placeholders
- [x] add a button instead of "contact you@dev.to"
- [x] add a checkbox "I'm the author of the podcast", and create an appropriate role while creating a podcast if it's checked
- [x] add `creator_id` and an association to podcasts to store who added a podcast
- [ ] improve the looks

## Related Tickets & Documents
Closes #3537

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Screenshot_2019-12-20 Podcasts - DEV(local) Community 👩‍💻👨‍💻

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -51,6 +51,7 @@ class User < ApplicationRecord
   has_many :webhook_endpoints, class_name: "Webhook::Endpoint", foreign_key: :user_id, inverse_of: :user, dependent: :delete_all
   has_many :user_blocks
   has_one :pro_membership, dependent: :destroy
+  has_many :created_podcasts, class_name: "Podcast", foreign_key: :creator_id, inverse_of: :creator, dependent: :nullify
```
Comment (rhymes) [inline app/models/user.rb]: I was going to ask "why not just .podcasts" but I can see why in the future we might want to have podcast editors or other podcasts tied to a users in different capacity

File: db/migrate/20191220120243_add_create_by_user_id_to_podcasts.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class AddCreateByUserIdToPodcasts < ActiveRecord::Migration[5.2]
+  def change
+    add_column :podcasts, :creator_id, :integer
+    add_index :podcasts, :creator_id
+    add_foreign_key :podcasts, :users, column: :creator_id, null: true
```
Comment (rhymes) [inline db/migrate/20191220120243_add_create_by_user_id_to_podcasts.rb]: yes :D

**Comment by lightalloy** [comment]
@rhymes when a podcast is suggested, it stays in the database as an unpublished one, so it won't be on the site and the episodes won't be fetched. Site admin can go into `/internal/podcasts` and make it published. In this case, the episodes will be fetched and the podcast will appear in `/pod`.
Currently, information about creators and podcasts' owners is just for admins and won't be displayed on a site. Full podcast ownership (like a user has a list of their podcasts on their dashboard, etc) is a part of the further feature development (but it's not a priority task right now)
I didn't fully get the meaning of "I'm the owner of someone else's podcast and I create it through this form". If you are an owner, it's your podcast, not someone else's. If you are not an owner, you won't check the checkbox and will be stored as a creator of the podcast, not an owner. A situation when a user claims that they are an owner of the podcast but they are not is another question, I think it could be handled with the help of the site admins.

**Comment by rhymes** [comment]
Sorry, I wasn't clear at all, what I meant is what you said here: 

> A situation when a user claims that they are an owner of the podcast but they are not is another question, I think it could be handled with the help of the site admins.

me claiming to be the owner even if I'm not. Agreed that this will be handled by site admins

**Comment by lightalloy** [comment]
Thanks for merging, @benhalpern

**Pull request merged**